app/models/pxe_server.rb

```diff
@@ -59,7 +59,7 @@ def synchronize_advertised_images_queue
   def sync_images
     sync_pxe_images
     sync_windows_images
-    clear_association_cache
+    reload
```

Comment:
See the test changes below... I think this has an unexpected change in behavior.  Not persisted records will fail reload and changed records (like the tests) will lose these changes on reload.

Maybe we should target the associations?

```
pxe_images.reload
pxe_menus.reload
```

Reply:
Forget it.  I don't like the test changes.  Throwing away in memory / not yet persisted records / changes is what we we originally needed.  Let me try reloading the associations if the object is persisted.